packages/expo-cli/src/commands/start/startAsync.ts

```diff
@@ -25,6 +26,13 @@ export async function actionAsync(projectRoot: string, options: NormalizedOption
 
   // Only validate expo in Expo Go contexts
   if (!options.devClient) {
+    const hasDevClientInstalled = isDevClientInstalledInProject(projectRoot);
```

Comment:
Wrap with the profiling method:
```suggestion
    const hasDevClientInstalled = profileMethod(isDevClientInstalledInProject)(projectRoot);
```

Then run with `EXPO_PROFILE=1 expod start`, this adds:

- new managed project: 0.722ms (not warned)
- new project with `yarn add expo-dev-client`: 1.873s (warned)
- full project (`expo/apps/native-component-list`): 3.759s (not warned)

This is too much time to delay the `expo start` command.